keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/MemoryCloudDriverCache.kt

```diff
@@ -105,15 +128,17 @@ class MemoryCloudDriverCache(
     }
 
   override fun availabilityZonesBy(account: String, vpcId: String, purpose: String, region: String): Set<String> =
-    availabilityZones.get("$account:$vpcId:$purpose:$region") {
-      runBlocking {
-        cloudDriver
-          .listSubnets("aws", DEFAULT_SERVICE_ACCOUNT)
-          .filter { it.account == account && it.vpcId == vpcId && it.purpose == purpose && it.region == region }
-          .map { it.availabilityZone }
-          .toSet()
-      }
-    }!!
+    runBlocking {
+      availabilityZones.get("$account:$vpcId:$purpose:$region",
+        asyncify {
+          cloudDriver
+            .listSubnets("aws", DEFAULT_SERVICE_ACCOUNT)
+            .filter { it.account == account && it.vpcId == vpcId && it.purpose == purpose && it.region == region }
+            .map { it.availabilityZone }
+            .toSet()
+        }
+      ).await()
+    }
```

Comment:
Do we have to wrap the initial cache attempt  with `runBlocking`?

Reply:
Yeah, the returned value from the cache is now a `CompleteableFuture`, which we call [await] on (line 140). `await` is a suspend function, so it needs to bre wrapped.

[await]: https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-jdk8/kotlinx.coroutines.future/java.util.concurrent.-completion-stage/await.html